libs/core/core/hooks/use-msc-device-detached-effect.ts

```diff
@@ -40,6 +41,9 @@ const useHandleDevicesDetached = () => {
   const flashingInWaitingForBackButtonState = useSelector(
     selectIsFlashingInState(FlashingProcessState.WaitingForBackButton)
   )
+  const flashingInRestartingPhase = useSelector(
+    selectIsFlashingInState(FlashingProcessState.Restarting)
+  )
```

Comment:
Hey, I wanted to point out this code because I made a mistake in my previous comment. Maybe we could fix it together by using a selector with dynamic arguments instead of a factory? This way, we can avoid creating a new selector with each re-render, allowing it to retain the memorized values.

```suggestion
  const flashingInWaitingForBackButtonState = useSelector((state) => 
    selectIsFlashingInState(state, FlashingProcessState.WaitingForBackButton)
  )
  const flashingInRestartingPhase = useSelector((state) => 
    selectIsFlashingInState(state, FlashingProcessState.Restarting)
  )
```